plugin/types.go

```diff
@@ -46,12 +46,18 @@ type Entry interface {
 	id() string
 	setID(id string)
 	getTTLOf(op defaultOpCode) time.Duration
+	shortType() string
+	isSingleton() bool
+	// This is used by the plugin registry to mark plugin roots as singleton
+	// objects. This way, plugin authors don't have to do this themselves.
+	markSingleton()
 }
 
 // Parent is an entry with children. It will be represented as a directory in the Wash
 // filesystem.
 type Parent interface {
 	Entry
+	ChildSchemas() []EntrySchema
```

Comment:
Reason for `ChildSchemas` is to force core plugin authors (at compile-time) to provide this info as I don't see a good reason why people shouldn't document their plugins while writing them.